HTAN1 Sunset: Revoke bucket permissions for non DCC members

aditigopalan opened the pull request:
### Overview
Updates HTAN DCC bucket permissions: adds a new user to S3UserARNs across all HTAN DCC buckets and revokes access for non-essential DCC personnel as part of HTAN 1 sunset.

### Changes

**1. Add aditi.gopalan to S3UserARNs for HTAN DCC buckets and inodb to DenyDeleteARNs for MSK** (`8da4d27`)
- Added `aditi.gopalan@sagebase.org` ARN to `S3UserARNs` in all HTAN DCC bucket configurations:
  - htan-dcc-casi-tnp.yaml
  - htan-dcc-center-a.yaml
  - htan-dcc-chop.yaml
  - htan-dcc-dfci.yaml
  - htan-dcc-duke.yaml
  - htan-dcc-hms.yaml
  - htan-dcc-msk.yaml
  - htan-dcc-ohsu.yaml
  - htan-dcc-pcapp.yaml
  - htan-dcc-stanford.yaml
  - htan-dcc-tma-tnp.yaml
  - htan-dcc-tnp-sardana.yaml
  - htan-dcc-vanderbilt.yaml
  - Added `inodb` user to `DenyDeleteARNs` in `htan-dcc-msk.yaml` to match the pattern used in `htan-dcc-center-a.yaml`

**2. HTAN 1 Sunset: Revoking non-DCC bucket access** (`ad13d1a`)
- Revoked bucket access for non-essential DCC personnel as part of the HTAN 1 sunset process

### Testing
- All modified files passed pre-commit checks (), no changes to those files

aditigopalan committed:
HTAN 1 Sunset: Revoking non-DCC bucket access

Revoking bucket access for any non essential DCC personnel

aditigopalan committed:
Add aditi.gopalan to S3UserARNs for HTAN DCC buckets and inodb to DenyDeleteARNs for MSK

- Add aditi.gopalan@sagebase.org ARN to S3UserARNs in all HTAN DCC bucket configs
- Add inodb user to DenyDeleteARNs in htan-dcc-msk.yaml to match center-a pattern

aditigopalan commented:
fyi @adamjtaylor

adamjtaylor commented on config/prod/htan-dcc-dfci.yaml:
Remove bill from deny delete

adamjtaylor commented on config/prod/htan-dcc-center-a.yaml:
Remove bill from deny delete

adamjtaylor commented on config/prod/htan-dcc-chop.yaml:
Remove Bill from deny delete

adamjtaylor commented on config/prod/htan-dcc-duke.yaml:
Remove from deny delete

adamjtaylor commented on config/prod/htan-dcc-duke.yaml:
Remove from deny delete

adamjtaylor reviewed (changes requested):
Where we remove users from `S3UserARNs` we also should remove from `DenyDeleteARNs`

Have added specific comments for some users but need to do for all

aditigopalan committed:
Remove users from DenyDeleteARNs that were removed from S3UserARNs

As part of HTAN 1 Sunset cleanup, remove users from DenyDeleteARNs that
were previously removed from S3UserARNs to maintain consistency.

Removed users:
- bill.clifford (all 13 HTAN DCC buckets)
- jmuhlich (hms, tma-tnp, tnp-sardana)
- creason (ohsu, tma-tnp, tnp-sardana)
- jchan, roses3, moormana, alejandro (msk)
- stanford-htan-uploader, stanford-htan-thomas (stanford, tnp-sardana)
- abs33@duke, mrl17@duke, svennam (duke)

adamjtaylor reviewed (approved):
LGTM! Thank you!

adamjtaylor commented:
Pre commit is failing, but oddly not on the buckets the PR changes, suggesting that these had been forced through in an earlier PR.

These are cost center errors, some for HTAN so we should check.

>  - config/prod/psychencode-migration-bucket.yaml: "UMass PECDCC / 106000" is not a valid CostCenter
> - config/prod/htan2-prostate-syn63298068.yaml: "DFCI HTAN / 120100" is not a valid CostCenter
> - config/prod/htan2-myeloma-syn63298063.yaml: "DFCI HTAN / 120100" is not a valid CostCenter
> - config/prod/htan2-gastric-syn63298051.yaml: "DFCI HTAN / 120100" is not a valid CostCenter
> - config/prod/htan2-crc-syn63298073.yaml: "DFCI HTAN / 120100" is not a valid CostCenter
> - config/prod/htan2-ovarian-syn63298044.yaml: "DFCI HTAN / 120100" is not a valid CostCenter
> - config/prod/cmc-migration-bucket.yaml: "UMass PECDCC / 106000" is not a valid CostCenter
> - config/prod/htan2-lymphoma-syn63298076.yaml: "DFCI HTAN / 120100" is not a valid CostCenter
> - config/prod/htan2-skin-syn63298054.yaml: "DFCI HTAN / 120100" is not a valid CostCenter
> - config/prod/htan2-glioma-syn63298048.yaml: "DFCI HTAN / 120100" is not a valid CostCenter
> - config/prod/htan2-pediatric-syn63298059.yaml: "DFCI HTAN / 120100" is not a valid CostCenter
> - config/prod/htan2-pancreatic-syn63298065.yaml: "DFCI HTAN / 120100" is not a valid CostCenter
> - config/prod/nda-migration.yaml: "UMass PECDCC / 106000" is not a valid CostCenter

adamjtaylor commented:
The valid cost center for HTAN apears to be `HTAN-DFCI / 120100` rather than `DFCI HTAN / 120100`

adamjtaylor commented:
I think I am following the pre-commit in that valid codes can be got from `curl https://mips-api.finops.sageit.org/tags` and includes `DFCI HTAN Renewal / 120100` as the only valid live HTAN code. The Deprecated Tags json in the repo includes old valid tags to maintain compatabilty,  including `HTAN-DFCI / 120100`

@aditigopalan I suggest we update all HTAN1 and HTAN2 buckets to use `DFCI HTAN Renewal / 120100` as the CostCenter` in this PR.

adamjtaylor commented:
This will still have the pre-commit fail on the other codes that are not within Cancer Bio's portfolio

aditigopalan committed:
Update CostCenter to 'DFCI HTAN Renewal / 120100' for all HTAN1 and HTAN2 buckets

Update CostCenter tag from 'HTAN-DFCI / 120100' and 'DFCI HTAN / 120100'
to 'DFCI HTAN Renewal / 120100' for:
- All 13 HTAN1 DCC bucket configurations
- All 10 HTAN2 bucket configurations
- HTAN2 testing bucket
- HTAN KMS key configuration

adamjtaylor reviewed (approved):
Super!

adamjtaylor commented:
Tagging @danlu1 and @thomasyu888 as owners of the buckets that have outstanding pre-commit errors

```
- config/prod/psychencode-migration-bucket.yaml: "UMass PECDCC / 106000" is not a valid CostCenter
- config/prod/cmc-migration-bucket.yaml: "UMass PECDCC / 106000" is not a valid CostCenter
- config/prod/nda-migration.yaml: "UMass PECDCC / 106000" is not a valid CostCenter
```

thomasyu888 commented:
Quoting a reply from Joni from slack https://sagebionetworks.slack.com/archives/CEFQD0KU1/p1764180601175379?thread_ts=1764176530.351069&cid=CEFQD0KU1: @aditigopalan 

> As a quick workaround, the tag value can be added to the list of [deprecated codes](https://github.com/Sage-Bionetworks/scicomp-provisioner/blob/master/DeprecatedProgramCodes.json).

adamjtaylor commented:
@aditigopalan lets add that code causing the error to the file as @thomasyu888 and @ConsoleCatzirl suggest.

aditigopalan committed:
Add UMass PECDCC / 106000 to DeprecatedProgramCodes.json

aditigopalan commented:
Done, this is ready to merge now
